lib/pdksync.rb

```diff
@@ -44,10 +47,147 @@ module PdkSync
+    jenkins_username: Constants::JENKINS_USERNAME,
+    jenkins_password: Constants::JENKINS_PASSWORD,
+    jenkins_api_endpoint: Constants::JENKINS_API_ENDPOINT
+  }
+
+  # convert duration from milliseconds to format h m s ms
+  def self.get_duration_hrs_and_mins(milliseconds)
+    return '' unless milliseconds
+    hours, milliseconds   = milliseconds.divmod(1000 * 60 * 60)
+    minutes, milliseconds = milliseconds.divmod(1000 * 60)
+    seconds, milliseconds = milliseconds.divmod(1000)
+    "#{hours}h #{minutes}m #{seconds}s #{milliseconds}ms"
+  end
+
+  # jenkins report
+  def self.jenkins_report_analisation(github_repo, build_id)
+    job_name = "forge-module_#{github_repo}_init-manual-parameters_adhoc"
+    adhoc_urls = []
+    # add init build url to ad-hoc urls
+    adhoc_urls.push("https://jenkins-master-prod-1.delivery.puppetlabs.net/job/#{job_name}")
+
+    # create adhoc_url slist
+    adhoc_urls.each do |url|
+      build_job_data = HTTParty.get("#{url}/api/json").parsed_response
+      downstream_job = build_job_data['downstreamProjects']
+      break if downstream_job.empty?
+      downstream_job.each do |item|
+        next if item.nil?
+        adhoc_urls.push(item['url']) unless item['url'].nil? && item['url'].include?('skippable_adhoc')
+      end
+    end
+    # remove duplicates and sort it
+    adhoc_urls = adhoc_urls.uniq
+    adhoc_urls = adhoc_urls.sort_by { |url| HTTParty.get("#{url}/api/json").parsed_response['fullDisplayName'].scan(%r{[0-9]{2}\s}).first.to_i }
+
+    # analyse each build result
+    # status, execution time, full-log in case of failures
+    adhoc_urls.each do |url|
+      puts '-' * 100
+      # next if skipped
+      current_build_data = HTTParty.get("#{url}/api/json").parsed_response
+      puts "Skipped build: #{current_build_data['fullDisplayName']}" if url.include?('skippable_adhoc') ||
+                                                                        current_build_data['color'] == 'notbuilt' ||
+                                                                        current_build_data['fullDisplayName'].downcase.include?('skipped')
+      next if url.include?('skippable_adhoc') || current_build_data['color'] == 'notbuilt'
+      next if current_build_data['fullDisplayName'].downcase.include?('skipped')
+      puts url
+      get_data_build(url, build_id)
+    end
+  end
+
+  # for each build from adhoc jobs, get data
+  # if multiple platforms under current url, get data for each platform
+  def self.get_data_build(url, build_id)
+    current_build_data = HTTParty.get("#{url}/api/json").parsed_response
+    puts "Build name=#{current_build_data['fullDisplayName']}"
+
+    if current_build_data['activeConfigurations'].nil?
+      # builds don't have the sabe build_id. That's why just the init build will be identified by id, rest of them by lastBuild
+      current_build_url = "#{url}/#{build_id}/api/json" if url.include?('init-manual-parameters_adhoc')
```

Comment:
You are double assigning here and this is confusing.  

Instead of 

```
current_build_url = "#{url}/#{build_id}/api/json" if url.include?('init-manual-parameters_adhoc')
current_build_url = "#{url}lastBuild/api/json" unless url.include?('init-manual-parameters_adhoc')
```

Use

```
current_build_url = url.include?('init-manual-parameters_adhoc') ? "#{url}/#{build_id}/api/json" : "#{url}/lastBuild/api/json"
```